src/System.Memory/tests/ReadOnlyBuffer/ReadOnlyBufferFactory.cs

```diff
@@ -0,0 +1,133 @@
+
+        public abstract ReadOnlyBuffer<byte> CreateOfSize(int size);
+        public abstract ReadOnlyBuffer<byte> CreateWithContent(byte[] data);
+
+        public ReadOnlyBuffer<byte> CreateWithContent(string data)
+        {
+            return CreateWithContent(Encoding.ASCII.GetBytes(data));
+        }
+
+        internal class ArrayTestBufferFactory : ReadOnlyBufferFactory
+        {
+            public override ReadOnlyBuffer<byte> CreateOfSize(int size)
+            {
+                return new ReadOnlyBuffer<byte>(new byte[size + 20], 10, size);
+            }
+
+            public override ReadOnlyBuffer<byte> CreateWithContent(byte[] data)
+            {
+                var startSegment = new byte[data.Length + 20];
+                System.Array.Copy(data, 0, startSegment, 10, data.Length);
+                return new ReadOnlyBuffer<byte>(startSegment, 10, data.Length);
+            }
+        }
+
+        internal class MemoryTestBufferFactory : ReadOnlyBufferFactory
+        {
+            public override ReadOnlyBuffer<byte> CreateOfSize(int size)
+            {
+                return CreateWithContent(new byte[size]);
+            }
+
+            public override ReadOnlyBuffer<byte> CreateWithContent(byte[] data)
+            {
+                var startSegment = new byte[data.Length + 20];
+                System.Array.Copy(data, 0, startSegment, 10, data.Length);
+                return new ReadOnlyBuffer<byte>(new Memory<byte>(startSegment, 10, data.Length));
+            }
+        }
+
+        internal class SingleSegmentTestBufferFactory: ReadOnlyBufferFactory
+        {
+            public override ReadOnlyBuffer<byte> CreateOfSize(int size)
+            {
+                return CreateWithContent(new byte[size]);
+            }
+
+            public override ReadOnlyBuffer<byte> CreateWithContent(byte[] data)
+            {
+                return CreateSegments(data);
+            }
+        }
+
+        internal class BytePerSegmentTestBufferFactory: ReadOnlyBufferFactory
+        {
+            public override ReadOnlyBuffer<byte> CreateOfSize(int size)
+            {
+                return CreateWithContent(new byte[size]);
+            }
+
+            public override ReadOnlyBuffer<byte> CreateWithContent(byte[] data)
+            {
+                var segments = new List<byte[]>();
+
+                segments.Add(System.Array.Empty<byte>());
+                foreach (var b in data)
+                {
+                    segments.Add(new [] { b });
+                    segments.Add(System.Array.Empty<byte>());
+                }
+
+                return CreateSegments(segments.ToArray());
+            }
+        }
+
+        public static ReadOnlyBuffer<byte> CreateSegments(params byte[][] inputs)
+        {
+            if (inputs == null || inputs.Length == 0)
+            {
+                throw new InvalidOperationException();
```

Comment:
should this use ThrowHelper.ThrowInvalidOperationException..

Reply:
Test code.